ipn/ipnlocal/local.go

```diff
@@ -3114,6 +3114,19 @@ func (b *LocalBackend) SetUseExitNodeEnabled(v bool) (ipn.PrefsView, error) {
 	return b.editPrefsLockedOnEntry(mp, unlock)
 }
 
+// MaybeClearAppConnector clears the routes from any AppConnector if
+// AdvertiseRoutes has been set in the MaskedPrefs.
+func (b *LocalBackend) MaybeClearAppConnector(mp *ipn.MaskedPrefs) error {
+	var err error
+	if b.appConnector != nil && mp.AdvertiseRoutesSet {
+		err = b.appConnector.ClearRoutes()
+		if err != nil {
+			b.logf("appc: clear routes error: %v", err)
```

Comment:
Hmm, will the error get logged elsewhere too? Might be redundant. Modifying `err` with a `fmt.Errorf` may be reasonable though.

Reply:
We'll return the error as the body of a 500 so it seems likely to me that under most configurations it'll get logged somewhere yes. I don't think it'll get logged often and it seems nice to me to also log it here where it happened, not just as the byproduct of likely http logging.